cmd/state/main.go

```diff
@@ -219,6 +220,10 @@ func run(args []string, isInteractive bool, cfg *config.Instance, out output.Out
 	project.RegisterExpander("mixin", project.NewMixin(auth).Expander)
 	project.RegisterExpander("secrets", project.NewSecretPromptingExpander(secretsapi.Get(), prompter, cfg, auth))
 
+	if pj != nil {
+		projectmigration.PromptToMigrateIfNecessary(pj, prompter, out)
+	}
```

Comment:
I'd rather see this addressed when we try to access the local commit. Because otherwise we'll be doing this check on commands that don't even use the yaml file. And it could be problematic for commands such as `state _prepare`.